server/handlers/project_command_output_handler.go

```diff
@@ -172,40 +184,65 @@ func (p *AsyncProjectCommandOutputHandler) writeLogLine(pull string, line string
 	}
 
 	p.projectOutputBuffersLock.Lock()
-	if p.projectOutputBuffers[pull] == nil {
-		p.projectOutputBuffers[pull] = []string{}
+	if p.projectOutputBuffers[jobID] == nil {
+		p.projectOutputBuffers[jobID] = []string{}
 	}
-	p.projectOutputBuffers[pull] = append(p.projectOutputBuffers[pull], line)
+	p.projectOutputBuffers[jobID] = append(p.projectOutputBuffers[jobID], line)
 	p.projectOutputBuffersLock.Unlock()
 }
 
 //Remove channel, so client no longer receives Terraform output
-func (p *AsyncProjectCommandOutputHandler) Deregister(pull string, ch chan string) {
-	p.logger.Debug("Removing channel for %s", pull)
+func (p *AsyncProjectCommandOutputHandler) Deregister(jobID string, ch chan string) {
+	p.logger.Debug("Removing channel for %s", jobID)
 	p.receiverBuffersLock.Lock()
-	delete(p.receiverBuffers[pull], ch)
+	delete(p.receiverBuffers[jobID], ch)
 	p.receiverBuffersLock.Unlock()
 }
 
-func (p *AsyncProjectCommandOutputHandler) GetReceiverBufferForPull(pull string) map[chan string]bool {
-	return p.receiverBuffers[pull]
+func (p *AsyncProjectCommandOutputHandler) GetReceiverBufferForPull(jobID string) map[chan string]bool {
+	return p.receiverBuffers[jobID]
 }
 
-func (p *AsyncProjectCommandOutputHandler) GetProjectOutputBuffer(pull string) []string {
-	return p.projectOutputBuffers[pull]
+func (p *AsyncProjectCommandOutputHandler) GetProjectOutputBuffer(jobID string) []string {
+	return p.projectOutputBuffers[jobID]
 }
 
-func (p *AsyncProjectCommandOutputHandler) CleanUp(pull string) {
-	p.projectOutputBuffersLock.Lock()
-	delete(p.projectOutputBuffers, pull)
-	p.projectOutputBuffersLock.Unlock()
+func (p *AsyncProjectCommandOutputHandler) CleanUp(jobContext models.JobContext) {
```

Comment:
ah this is the context I was missing when I was asking why we need to maintain a reverse mapping to lookup job id for a specific commit.

Reply:
Yup, I didn't notice it until I started working on it haha